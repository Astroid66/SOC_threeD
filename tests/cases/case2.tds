int main()
{
	int num1 = 0;
	int num2, num3 = 42;
	const char* str1 = "hello"; // This is a C style const string.
	return 0;
}