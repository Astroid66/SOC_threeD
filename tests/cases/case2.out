TYPE int
IDENT main
OPARA (
CPARA )
OBRACE {
TYPE int
IDENT num1
OP_EQ =
INT_LITERAL 0
SEMICOLON ;
TYPE int
IDENT num2
OP_COMMA ,
IDENT num3
OP_EQ =
INT_LITERAL 42
SEMICOLON ;
CONST const
TYPE char*
IDENT str1
OP_EQ =
STRING_LITERAL "hello"
SEMICOLON ;
RET return
INT_LITERAL 0
SEMICOLON ;
CBRACE }